api/v1alpha1/envoygateway_helpers.go

```diff
@@ -85,6 +86,11 @@ func (e *EnvoyGateway) NamespaceMode() bool {
 		len(e.Provider.Kubernetes.Watch.Namespaces) > 0
 }
 
+// DefaultLeaderElection returns a new LeaderElection with default configuration parameters.
```

Comment:
decision to make here, if we are enabling by default
* do we do it via helm (instantiate the field) or in process
* how do we disable it ?

good to make a call after looking around to see what the current state of existing CNCF projects is
does the average admin find default leaderElection stanza to be overwhelming or irrelevant

Reply:
We can configure this at the helm level, allowing users to determine how Envoy should initiate upon deployment. Alternatively, we could adhere to the Envoy resource spec, in this case we need to decide on how we startup, probably with leader election enabled by default and possibly alter its mode of operation once the resource is processed. 

i'm uncertain whether disabling Envoy's leader election during runtime is interesting? or we can assume this is an eg deployment setting.
 
Istio leader election is enabled by default as per:

ENABLE_LEADER_ELECTION	Boolean	true	If enabled (default), starts a leader election client and gains leadership before executing controllers. If false, it assumes that only one instance of istiod is running and skips leader election.

https://github.com/istio/istio/pull/41141/files

Reply:
* lets define `envoyGateway.leaderElection: {}` in helm for now
* lets disable leader election if `envoyGateway.leaderElection` is `nil`

Reply:
or we add a explicit `disabled *bool` flag in the `leaderElection` struct, will bring this up in the community meeting tomorrow

Reply:
we discussed this in the community meeting yesterday, and we agreed that lets enable it by default, and add an explicit `Disable bool` knob to disable it